processor/trackingplan.go

```diff
@@ -0,0 +1,250 @@
+			marshalledParams = []byte(`{"error": "Processor failed to marshal params"}`)
+		}
+		newFailedJob := jobsdb.JobT{
+			UUID:         id,
+			EventPayload: payload,
+			Parameters:   marshalledParams,
+			CreatedAt:    time.Now(),
+			ExpireAt:     time.Now(),
+			UserID:       dropEvent.Metadata.RudderID,
+		}
+		failedEventsToStore = append(failedEventsToStore, &newFailedJob)
+
+	}
+	return eventsToTransform, failedEventsToStore
+}
+
+func fetchEventConfig(sourceTpConfig map[string]interface{}, eventType string) map[string]interface{} {
+	emptyMap := map[string]interface{}{}
+	_, eventSpecificConfigPresent := sourceTpConfig[eventType]
+	if !eventSpecificConfigPresent {
+		return emptyMap
+	}
+	eventSpecificConfig, castOk := sourceTpConfig[eventType].(map[string]interface{})
+	if !castOk {
+		pkgLogger.Errorf("config not parseable for %s", eventType)
+		return emptyMap
+	}
+	return eventSpecificConfig
+}
+
+func reportViolations(validateEvent *transformer.TransformerResponseT) {
+	if len(validateEvent.ValidationErrors) == 0 {
+		return
+	}
+	validationErrors := validateEvent.ValidationErrors
+	output := validateEvent.Output
+
+	pkgLogger.Errorf("%d errors reported", len(validationErrors))
+	pkgLogger.Error(validationErrors)
+
+	eventContext, castOk := output["context"].(map[string]interface{})
+	if castOk {
+		eventContext["violationErrors"] = validationErrors
+	}
+}
+
+func (proc *HandleT) validateEvents(groupedEventsBySourceID map[string][]transformer.TransformerEventT, eventsByMessageID map[string]types.SingularEventWithReceivedAt) map[string][]transformer.TransformerEventT {
+	//validating with the tp here for every sourceID
+	var validatedEventsBySourceID = make(map[string][]transformer.TransformerEventT)
+	for srcId, eventList := range groupedEventsBySourceID {
+		isTpExists := eventList[0].Metadata.TrackingPlanId != ""
+		if !isTpExists {
+			// pass on the jobs for transformation(User,Dest)
+			validatedEventsBySourceID[srcId] = make([]transformer.TransformerEventT, 0)
+			validatedEventsBySourceID[srcId] = append(validatedEventsBySourceID[srcId], eventList...)
+			continue
+		}
+		startedAt := time.Now()
+		response := proc.transformer.Validate(eventList, integrations.GetTrackingPlanValidationURL(), userTransformBatchSize)
+		endedAt := time.Now()
+		timeTaken := endedAt.Sub(startedAt).Seconds()
+		fmt.Println(timeTaken)
+
+		commonMetaData := transformer.MetadataT{
+			SourceID:       srcId,
+			SourceType:     eventList[0].Metadata.SourceType,
+			SourceCategory: eventList[0].Metadata.SourceCategory,
+		}
+
+		//handling incompatible version types between rudder-server and transformer
+		filteredFailedEvents := []transformer.TransformerResponseT{}
+		for _, event := range response.FailedEvents {
+			//handling case when rudder-transformer is missing validation end-point
+			//passing on the events as successfull further forward to user/dest transformation
+			if event.StatusCode == 404 {
```

Comment:
As tp is still in initial phase, forward all 400 events too for now, 
Once when fully developed(like supporting multiple jsonschema versions etc), events can be dropped to proc_error

Reply:
Done.